avm/res/sql/server/vulnerability-assessment/main.bicep

```diff
@@ -0,0 +1,64 @@
+metadata name = 'Azure SQL Server Vulnerability Assessments'
+metadata description = 'This module deploys an Azure SQL Server Vulnerability Assessment.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the vulnerability assessment.')
+param name string
+
+@description('Conditional. The Name of SQL Server. Required if the template is used in a standalone deployment.')
+param serverName string
+
+@description('Optional. Recurring scans state.')
+param recurringScansIsEnabled bool = false
+
+@description('Optional. Specifies that the schedule scan notification will be is sent to the subscription administrators.')
+param recurringScansEmailSubscriptionAdmins bool = false
+
+@description('Optional. Specifies an array of email addresses to which the scan notification is sent.')
+param recurringScansEmails array = []
+
+@description('Required. A blob storage to hold the scan results.')
+param storageAccountResourceId string
+
+@description('Optional. Use Access Key to access the storage account. The storage account cannot be behind a firewall or virtual network. If an access key is not used, the SQL Server system assigned managed identity must be assigned the Storage Blob Data Contributor role on the storage account.')
+param useStorageAccountAccessKey bool = false
+
+@description('Optional. Create the Storage Blob Data Contributor role assignment on the storage account. Note, the role assignment must not already exist on the storage account.')
+param createStorageRoleAssignment bool = true
+
+resource server 'Microsoft.Sql/servers@2022-05-01-preview' existing = {
+  name: serverName
+}
+
+// Assign SQL Server MSI access to storage account
+module storageAccount_sbdc_rbac 'modules/nested_storageRoleAssignment.bicep' = if (!useStorageAccountAccessKey && createStorageRoleAssignment) {
+  name: '${server.name}-sbdc-rbac'
+  scope: resourceGroup(split(storageAccountResourceId, '/')[4])
```

Comment:
Unless there is a rule that the SA must be in the same subscription, we should probably change this to

```suggestion
  scope: resourceGroup(split(storageAccountResourceId, '/')[2], split(storageAccountResourceId, '/')[4])
```